packages/consensus/src/types/utxo_set.cairo

```diff
@@ -50,3 +52,92 @@ pub impl UtxoSetImpl of UtxoSetTrait {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{UtxoSet, UtxoSetTrait, OutPoint};
+    use consensus::types::transaction::TxOut;
+
+    #[test]
+    fn test_utreexo_add1() {
+        let mut utxo_set: UtxoSet = UtxoSetTrait::new(Default::default());
+
+        // https://learnmeabitcoin.com/explorer/tx/b1fea52486ce0c62bb442b530a3f0132b826c74e473d1f2c220bfa78111c5082#input-0
+        // coinbase outpoint
+        let outpoint1 = OutPoint {
+            txid: 0x0000000000000000000000000000000000000000000000000000000000000000_u256.into(),
+            vout: 4294967295,
+            data: TxOut { value: 0, pk_script: @"0x", cached: false },
+            block_height: 0,
+            block_time: 0,
+            is_coinbase: false
+        };
+
+        // https://learnmeabitcoin.com/explorer/tx/f4184fc596403b9d638783cf57adfe4c75c605f6356fbc91338530e9831e9e16#input-0
+        let outpoint2 = OutPoint {
+            txid: 0x0437cd7f8525ceed2324359c2d0ba26006d92d856a9c20fa0241106ee5a597c9_u256.into(),
+            vout: 0,
+            data: TxOut {
+                // https://learnmeabitcoin.com/explorer/tx/0437cd7f8525ceed2324359c2d0ba26006d92d856a9c20fa0241106ee5a597c9#output-0
+                value: 5000000000,
+                pk_script: @"0x410411db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5cb2e0eaddfb84ccf9744464f82e160bfa9b8b64f9d4c03f999b8643f656b412a3ac",
+                cached: false
+            },
+            block_height: 9,
+            block_time: 1231473279,
+            is_coinbase: true
+        };
+
+        utxo_set.add(outpoint1);
```

Comment:
It's a bit hard to verify, can you please change it so that we can observe who the roots array evolves? E.g. adding one leave -> got one root, add one more -> first one is empty, second is set, etc